tests/ui/fallible_impl_from.stderr

```diff
@@ -40,6 +40,16 @@ LL |             panic!();
    |             ^^^^^^^^^
    = note: this error originates in the macro `$crate::panic::panic_2015` (in Nightly builds, run with -Z macro-backtrace for more info)
 
+error: only a `panic!` in `if`-then statement
+  --> $DIR/fallible_impl_from.rs:28:9
+   |
+LL | /         if i != 42 {
+LL | |             panic!();
+LL | |         }
+   | |_________^ help: try: `assert!(!i != 42, "explicit panic");`
+   |
+   = note: `-D clippy::if-then-panic` implied by `-D warnings`
+
```

Comment:
Can you add `#![allow(clippy::if_then_panic)]` and remove these warnings in this test file? It looks a little confusing.

Reply:
If this suggestion even valid?

`!i != 42` is `(!i) != 42` which is not what is intended to be suggested here. It should be `!(i != 42)`.